pandas/tests/io/excel/test_readers.py

```diff
@@ -148,6 +149,33 @@ def parser(self, *args, **kwargs):
             expected = expected_defaults[read_ext[1:]]
         assert result == expected
 
+    def test_engine_kwargs(self, read_ext, engine):
+        # GH#52214
+        expected_defaults = {
+            "xlsx": {"foo": "abcd"},
+            "xlsm": {"foo": 123},
+            "xlsb": {"foo": "True"},
+            "xls": {"foo": True},
+            "ods": {"foo": "abcd"},
+        }
+
+        msg = re.escape(r"load_workbook() got an unexpected keyword argument 'foo'")
+
+        if read_ext[1:] == "xls" or read_ext[1:] == "xlsb":
+            msg = re.escape(r"open_workbook() got an unexpected keyword argument 'foo'")
+
+        elif read_ext[1:] == "ods":
+            msg = re.escape(r"load() got an unexpected keyword argument 'foo'")
```

Comment:
the extra newlines makes this hard to read, how about
```python
if read_ext[1:] == "xls" or read_ext[1:] == "xlsb":
    msg = ...
elif read_ext[1:] == "ods":
    msg = ...
else:
    msg = ...
```
?

Reply:
Fixed!